File: data/derived_cot/rq1_traces/translation/r1/cpp_to_py/code_output/JSONProcessor.py
import json

class JSONProcessor:
    def read_json(self, file_path):
        try:
            with open(file_path, 'r') as f:
                data = json.load(f)
        except OSError:
            return 0, None
        except Exception:
            return -1, None
        
        if data is None:
            return -1, None
            
        return 1, data

    def write_json(self, data, file_path):
        try:
            with open(file_path, 'w') as f:
                json.dump(data, f, indent=4)
        except Exception:
            return -1
        return 1

    def process_json(self, file_path, remove_key):
        result, data = self.read_json(file_path)
        if result != 1:
            return 0
            
        if isinstance(data, dict) and remove_key in data:
            del data[remove_key]
            return self.write_json(data, file_path)
        else:
            return 0

File: data/derived_cot/rq3_prompting/results/translation/qwen/py_to_cpp/code_output/JSONProcessor.cpp
#include <variant>
#include <filesystem>
#include <fstream>
#include <string>
#include <nlohmann/json.hpp>

using namespace nlohmann;

class JSONProcessor {
public:
    std::variant<int, json> read_json(const std::string& file_path) {
        if (!std::filesystem::exists(file_path)) {
            return 0;
        }
        std::ifstream i(file_path);
        if (!i.is_open()) {
            return -1;
        }
        json j;
        i >> j;
        if (i.fail()) {
            return -1;
        }
        return j;
    }

    int write_json(const json& data, const std::string& file_path) {
        std::ofstream o(file_path);
        o << data;
        if (o.fail()) {
            return -1;
        }
        return 1;
    }

    int process_json(const std::string& file_path, const std::string& remove_key) {
        auto result = read_json(file_path);
        if (std::holds_alternative<int>(result)) {
            return 0;
        }
        json& data = std::get<json>(result);
        if (data.find(remove_key) != data.end()) {
            data.erase(remove_key);
            write_json(data, file_path);
            return 1;
        }
        return 0;
    }
};